contracts/deposit-contracts/contract/src/main.rs

```diff
@@ -0,0 +1,214 @@
+        .into_uref()
+        .unwrap_or_revert()
+        .with_access_rights(AccessRights::READ_ADD_WRITE);
+    system::transfer_from_purse_to_purse(deposit_purse, destination_purse, amount, None)
+        .unwrap_or_revert();
+}
+
+#[no_mangle]
+pub extern "C" fn change_security() {
+    sec_check(vec![SecurityBadge::Admin]);
+    let admin_list: Option<Vec<Key>> =
+        get_optional_named_arg_with_user_errors(ADMIN_LIST, DepositError::InvalidAdminList);
+    // construct a new admin list from runtime arg
+    let mut badge_map: BTreeMap<Key, SecurityBadge> = BTreeMap::new();
+    if let Some(admin_list) = admin_list {
+        for account_key in admin_list {
+            badge_map.insert(account_key, SecurityBadge::Admin);
+        }
+    }
+    // remove the caller from the admin list
+    let caller = get_immediate_caller().unwrap_or_revert();
+    badge_map.remove(&caller);
+    security::change_sec_badge(&badge_map);
+}
+
+#[no_mangle]
+pub extern "C" fn call() {
+    let caller = Key::from(runtime::get_caller());
+
+    let admin_list: Option<Vec<Key>> =
+        get_optional_named_arg_with_user_errors(ADMIN_LIST, DepositError::InvalidAdminList);
+
+    let entry_points = {
+        let mut entry_points = EntryPoints::new();
+        entry_points.add_entry_point(entry_points::init());
+        entry_points.add_entry_point(entry_points::get_purse());
+        entry_points.add_entry_point(entry_points::deposit());
+        entry_points.add_entry_point(entry_points::withdrawal());
+        entry_points.add_entry_point(entry_points::incr_last_processed_deposit_counter());
+        entry_points.add_entry_point(entry_points::change_security());
+        entry_points
+    };
+    let mut named_keys = NamedKeys::new();
+    let event_dict = storage::new_dictionary(KAIROS_DEPOSIT_EVENT_DICT).unwrap_or_revert();
+    named_keys.insert(KAIROS_DEPOSIT_EVENT_DICT.to_string(), event_dict.into());
+    let last_processed_deposit_counter = storage::new_uref(u64::from(0u8));
+
+    named_keys.insert(
+        KAIROS_LAST_PROCESSED_DEPOSIT_COUNTER.to_string(),
+        last_processed_deposit_counter.into(),
+    );
+
+    let most_recent_deposit_counter = storage::new_uref(u64::from(0u8));
+    named_keys.insert(
+        KAIROS_MOST_RECENT_DEPOSIT_COUNTER.to_string(),
+        most_recent_deposit_counter.into(),
+    );
+
+    let (contract_hash, _) = storage::new_contract(
+        entry_points,
+        Some(named_keys),
+        Some(KAIROS_DEPOSIT_CONTRACT.to_string()),
+        Some(KAIROS_DEPOSIT_CONTRACT_PACKAGE.to_string()),
+    );
+    let contract_hash_key = Key::from(contract_hash);
+    runtime::put_key(KAIROS_DEPOSIT_CONTRACT_NAME, contract_hash_key);
+
+    // runtime arguments for contract initialization
+    let mut init_args = runtime_args! {};
+    if let Some(mut admin_list) = admin_list {
+        // prepare runtime arguments
+        init_args.insert(ADMIN_LIST, admin_list).unwrap_or_revert();
+    }
```

Comment:
Does this mean if there was no admin_list set, the contract won't have one at all? I would probably not allow this and fail

Reply:
No, the caller is the default admin.

Reply:
But Andrzej wants the access control to be deleted alongside the withdrawal EP. So if I end up doing that then there will be no admin at all.

Reply:
@jonas089 I don't want to remove access control. I want to understand why you introduced it, and what are your intentions. Then we can talk whether it's necessary or not.